File: library/api/src/main/java/org/cru/godtools/api/model/GlobalActivityAnalytics.java
package org.cru.godtools.api.model;

import org.ccci.gto.android.common.jsonapi.annotation.JsonApiAttribute;
import org.ccci.gto.android.common.jsonapi.annotation.JsonApiType;

import java.util.Date;

import static org.cru.godtools.api.model.GlobalActivityAnalytics.JSON_TYPE_GLOBAL_ACTIVITY_ANALYTICS;

@JsonApiType(JSON_TYPE_GLOBAL_ACTIVITY_ANALYTICS)
public class GlobalActivityAnalytics {
    static final String JSON_TYPE_GLOBAL_ACTIVITY_ANALYTICS = "global-activity-analytics";
    private final String JSON_USERS = "users";
    public final String JSON_COUNTRIES = "countries";
    public final String JSON_LAUNCHES = "launches";
    public final String JSON_GOSPEL_PRESENTATIONS = "gospel-presentations";
    public final String TABLE_NAME_GLOBAL_ACTIVITY_ANALYTICS = "global_activity_analytics";

    @JsonApiAttribute(JSON_USERS)
    private int users;

    @JsonApiAttribute(JSON_COUNTRIES)
    private int countries;

    @JsonApiAttribute(JSON_LAUNCHES)
    private int launches;

    @JsonApiAttribute(JSON_GOSPEL_PRESENTATIONS)
    private int gospelPresentation;

    @JsonApiAttribute(TABLE_NAME_GLOBAL_ACTIVITY_ANALYTICS)
    private Date lastUpdated;

    public int getUsers() {
        return users;
    }

    public void setUsers(final int users) {
        this.users = users;
    }

    public int getCountries() {
        return countries;
    }

    public void setCountries(final int countries) {
        this.countries = countries;
    }

    public int getLaunches() {
        return launches;
    }

    public void setLaunches(final int launches) {
        this.launches = launches;
    }

    public int getGospelPresentation() {
        return gospelPresentation;
    }

    public void setGospelPresentation(final int gospelPresentation) {
        this.gospelPresentation = gospelPresentation;
    }

    public Date getLastUpdated() {
        return lastUpdated;
    }

    public void setLastUpdated(final Date lastUpdated) {
        this.lastUpdated = lastUpdated;
    }
}

File: library/global-activity-analytics-manger/src/main/java/org/cru/godtools/global/activity/analytics/manger/model/GlobalActivityAnalytics.kt
package org.cru.godtools.global.activity.analytics.manger.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import org.ccci.gto.android.common.jsonapi.annotation.JsonApiAttribute
import org.ccci.gto.android.common.jsonapi.annotation.JsonApiIgnore
import org.ccci.gto.android.common.jsonapi.annotation.JsonApiType
import java.util.Date

const val JSON_TYPE_GLOBAL_ACTIVITY_ANALYTICS = "global-activity-analytics"
private const val JSON_USERS = "users"
private const val JSON_COUNTRIES = "countries"
private const val JSON_LAUNCHES = "launches"
private const val JSON_GOSPEL_PRESENTATIONS = "gospel-presentations"
private const val TABLE_NAME_GLOBAL_ACTIVITY_ANALYTICS = "global_activity_analytics"

@Entity(tableName = TABLE_NAME_GLOBAL_ACTIVITY_ANALYTICS)
@JsonApiType(JSON_TYPE_GLOBAL_ACTIVITY_ANALYTICS)
class GlobalActivityAnalytics {
    @JsonApiIgnore
    @PrimaryKey
    var id = 1

    @JsonApiAttribute(name = JSON_USERS)
    var users = 0

    @JsonApiAttribute(name = JSON_COUNTRIES)
    var countries = 0

    @JsonApiAttribute(name = JSON_LAUNCHES)
    var launches = 0

    @JsonApiAttribute(name = JSON_GOSPEL_PRESENTATIONS)
    var gospelPresentation = 0

    @JsonApiIgnore
    var lastUpdated: Date? = null
}
